src/js/support/angular/timeline.line.directive.js

```diff
@@ -35,16 +36,43 @@ function($document, $window, $timeout, $log) {
 			}
 
 			// Check to see if filtering is enabled
-			if(null != attrs.sentioFilterFn) {
+			if (null != attrs.sentioFilterFn || attrs.sentioFilterState) {
 				timeline.filter(true);
 			}
+
+			// Store the filter state outside the scope as well as inside, to compare
+			var lastFilterState = null;
+
 			scope.$watch('filterFn', function(n, o){
 				timeline.filter().on('filterend', function(filterState){
 					$timeout(function(){
+						// Call the function callback
 						scope.filterFn({ filterState: filterState });
+
+						// Set the two-way-bound scope parameter
+						scope.filterState = filterState;
+
+						// Store the filter state locally so we can suppress updates on our own changes
+						lastFilterState = filterState;
 					});
 				});
 			});
+			scope.$watch('filterState', function(n, o) {
+				// If a filter was passed in and it is not the one we just set, do some updates
+				if (null != n && n !== lastFilterState) {
+
+					// If we're in the original format with 3 parameters, use the second two only
+					// TODO: We should go ahead and get rid of hte 3 parameter style
+					if (n.length > 2) {
+						// The first element indicates if we're disabled
+						if (n[0]) {
```

Comment:
I'm gonna leave this the way it is for now since the api isn't really clear on how you disable filtering. Should it be only if there is neither a filter function nor a bound filter variable?